common/src/main/java/com/wynntils/wynn/model/ShamanTotemModel.java

```diff
@@ -104,6 +106,61 @@ public void onTotemSpawn(AddEntityEvent e) {
                 3);
     }
 
+    @SubscribeEvent
+    public void onTimerSpawn(AddEntityEvent e) {
+        // We aren't looking for a new timer, skip
+        if (pendingTotem1VisibleId == null && pendingTotem2VisibleId == null && pendingTotem3VisibleId == null) return;
+
+        int entityId = e.getId();
+
+        // This timer is already bound to a totem but got respawned? skip
+        if (getBoundTotem(entityId) != null) return;
+
+        Entity possibleTimer = getBufferedEntity(entityId);
+        if (!(possibleTimer instanceof ArmorStand)) return;
+
+        // Given timerId is not a totem, make a new totem
+        List<ArmorStand> toCheck = McUtils.mc()
+                .level
+                .getEntitiesOfClass(
+                        ArmorStand.class,
+                        new AABB(
+                                possibleTimer.position().x - TOTEM_SEARCH_RADIUS,
+                                possibleTimer.position().y
+                                        - 0.3, // Don't modify this unless you are certain it is causing issues
+                                possibleTimer.position().z - TOTEM_SEARCH_RADIUS,
+                                possibleTimer.position().x + TOTEM_SEARCH_RADIUS,
+                                // (a LOT) more vertical radius required for totems casted off high places
+                                // This increased radius requirement increases as you cast from higher places and as the
+                                // server gets laggier
+                                possibleTimer.position().y + TOTEM_SEARCH_RADIUS * 5,
+                                possibleTimer.position().z + TOTEM_SEARCH_RADIUS));
+
+        for (ArmorStand as : toCheck) {
+            // Recreate location for each ArmorStand checked for most accurate coordinates
+            Location parsedLocation = new Location(as.position().x, as.position().y, as.position().z);
+            if (pendingTotem1VisibleId != null && as.getId() == pendingTotem1VisibleId) {
+                totem1 = new ShamanTotem(
+                        1, entityId, totem1.getVisibleEntityId(), -1, ShamanTotem.TotemState.ACTIVE, parsedLocation);
```

Comment:
`totem1 = new ShamanTotem(..., totem1.getVisibleEntityId())` looks a bit self-referential. I assume this means that you already have a totem1, but you want to update it with the entityId and parsedLocation of the timer entity?

I would suggest one of these two alternatives instead: (1 is simpler but I recommend 2)

1) Keep the overall idea, but extract `totem1.getVisibleEntityId()` into a variable first, like `originalTotemVisibleEntityId = totem1.getVisibleEntityId()` so you don't reference totem1 when creating the replacement.

2) Make ShamanTotem mutable on the entityId and parsedLocation field. I assume the totemState field is already mutable? If the idea is that exactly one ShamanTotem object corresponds to exactly one totem in the game, I think this is a better solution. You are just building it up piece by piece as you gather enough data, and until it has become ACTIVE, we can't be sure to have all data about it.